dask_grblas/expr.py

```diff
@@ -458,17 +555,105 @@ def __call__(self, *args, **kwargs):
         return updater[self.index]
 
     def update(self, obj):
-        self.parent[self.index] = obj
+        Assigner(Updater(self.parent), self.index).update(obj)
 
-    def __lshift__(self, obj):
-        self.update(obj)
+    def __lshift__(self, rhs):
+        self.update(rhs)
 
     @property
     def value(self):
         self._meta.value
         return self.new().value
 
 
+def _uniquify(index, obj):
+    # here we follow the SuiteSparse:GraphBLAS specification for
+    # duplicate index entries: ignore all but the last unique entry
+    rev_index = np.array(index)[::-1]
+    unique_indices, obj_indices = np.unique(rev_index, return_index=True)
+    if unique_indices.size < rev_index.size:
+        obj = obj[::-1].new()
+        obj = obj[obj_indices].new()
+        index = unique_indices
+    return index, obj
```

Comment:
Here we follow the SuiteSparse:GraphBLAS specification for duplicate index entries, i.e., ignore all duplicates but the last unique entry.  The original GraphBLAS C API says that duplicate indices lead to undefined behavior.